crates/red_knot_server/src/server/api/requests/diagnostic.rs

```diff
@@ -72,7 +72,7 @@ fn to_lsp_diagnostic(message: &str) -> Diagnostic {
     let words = message.split(':').collect::<Vec<_>>();
 
     let (range, message) = match words.as_slice() {
-        [_filename, line, column, message] => {
+        [_, _, line, column, message] | [_, line, column, message] => {
             let line = line.parse::<u32>().unwrap_or_default().saturating_sub(1);
```

Comment:
I think this should be temporary. This is because the virtual path would contain the entire URL (`untitled:Untitled-1`) which means the message would have an extra `:`